docs/source/ops/index.rst

```diff
@@ -0,0 +1,86 @@
+- Pruning may affect the behavior of Ledger API calls that allow to read data from the ledger: see the next sub-section for more information about API impacts.
+
+How the DAML Ledger API is affected
+-----------------------------------
+
+- Active data streams from the DAML Participant may abort and need to be re-established by the DAML application from a later offset than pruned, even if they are already streaming past it.
+- Requesting information at offsets that predate pruning, including from the ledger's start, will result in a ``NOT_FOUND`` gRPC error.
+  - As a consequence, after pruning, a DAML application must bootstrap from the Active Contract Service and a recent offset [3]_.
+
+Submission validation and DAML Ledger API endpoints that write to the ledger are generally not affected by pruning; an exception is that in-progress calls could abort while awaiting completion.
+
+Please refer to the :doc:`protobuf documentation of the API </app-dev/grpc/proto-docs>` for details about the ``prune`` operation itself and the behavior of other DAML Ledger API endpoints when pruning is being or has been performed.
+
+Other limitations
+-----------------
+
+- If the number of divulged contracts continually grows, the storage used by the participant node may still keep growing, even with regular pruning.
+- Pruning may be rejected even if the node is running correctly (for example, to preserve non-repudiation properties); in this case, the application might not be able to archive contracts containing PII or pruning of these contracts may not be possible; thus, actually deleting this PII may also be technically unfeasible.
+- Pruning may leave parties, packages, and configuration data on the participant node, even if they are no longer needed for transaction processing, and even if they contain PII [3]_.
+- Pruning does not move pruned information to cold storage but simply deletes pruned data; for this reason, it is advisable to back up the Participant Index DB before invoking pruning. See the next sub-section for more Participant Index DB-related advice before and after invoking `prune`.
+- Pruning is not selective but rather effectively truncates the ledger, removing the transactions and archived contracts at and before the pruning offset.
+
+.. [3] This might be improved in future versions.
+
+Pruning and the DAML Participant Index DB
+-----------------------------------------
+
+Activities to be carried out *before* invoking a pruning operation include:
+
+- Backing up the Participant Index DB, as pruning will not move information to cold storage but rather it will delete archived contracts and transactions earlier than or at the pruning offset.
+
+Activities to be carried out *after* invoking a pruning operation include:
+
+- On a PostreSQL Index DB, a manual `VACUUM FULL` command is needed for the DB to give back disk space to the OS.
```

Comment:
@fabiotudone-da : This is problematic, as it will block all inserts to the Index DB AFAIK.

I'd suggest we do not recommend this, but instead recommend using VACUUM and regular pruning to keep the disk file size bounded.

Reply:
@gerolf-da I think you suggested `VACUUM FULL`, what do you think? We could also suggest both, e.g. `VACUUM` to avoid downtime and the `FULL` variant when downtime is(/can be) planned to actually give back disk space to the OS.

Reply:
We probably should put a disclaimer that these are procedures that *can*/*should* be done before/after pruning, but that the user should consult with a "postgresql professional".

Reply:
Yes, we are definitely entering DB admin territory.

Reply:
Agreed that this is DB admin territory where multiple database server-specific actions may make sense (e.g. another one in case of Postgres: tune auto-vacuum).

Reply:
I added some more insights after our discussions and I think it is still relatively (enough) light-weight. Let me know WDYT.